Source/Dafny/AST/DafnyAst.cs

```diff
@@ -12434,6 +12434,7 @@ public class LitPattern : ExtendedPattern {
   }
 
   public class IdPattern : ExtendedPattern {
+    public bool HasParenthesis { get; }
```

Comment:
According to the documentation of `Arguments`, you should be able to have:
```suggestion
    public bool HasParenthesis => Arguments != null
```

That way you can remove `HasParenthesis` from the constructor

Reply:
`Arguments != null` means it's interpreted as a constructor by Dafny, but it may still not have any parenthesis in the syntax, which is exactly when this warning is shown.